### PR TITLE
Upgraded Borgmatic to version 1.9.0

### LINKED ISSUE #6195: [Package Update] borg to v1.4 and borgmatic to v1.8.13
### Software Name

borgbackup

### Brief Description

This is an update request to update borg from 1.2.6 to 1.4.0 to gain software (security) fixes and borgmatic from 1.8.2 to 1.8.13 to support additional monitoring services (like uptime kumo) and gain software (security) fixes.

Borgmatic: https://github.com/borgmatic-collective/borgmatic/releases

I guess it's a simple version update, as no breaking changes are listed either of the changelogs.

### Website

https://github.com/borgbackup/borg/releases

### Documentation

https://borgbackup.readthedocs.io/en/stable/changes.html#borg-1-2-x-to-1-4-x

### Build/Installation documentation

https://borgbackup.readthedocs.io/en/stable/

### Source code

https://github.com/borgbackup/borg/tree/1.4-maint

### OpenSource License

other (please specify below)

### Other License

BSD-3-Clause

### PR DESCRIPTION
**Pull request by tony1661**
## Description

- Borgmatic upgraded to version 1.9.0

Full list of changes can be found [here](https://github.com/borgmatic-collective/borgmatic/releases/tag/1.9.0).

New documentation can be found [here](https://github.com/SynoCommunity/spksrc/wiki/Borgmatic).

Fixes #6195

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [ ] Bug fix
- [ ] New Package
- [x] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

### CODE REVIEW

**Comment by tony1661** [comment]
I am ready to create documentation should this get merged. Just need to know where is the best place to do so.

**Comment by tony1661** [comment]
@hgy59 no reason. I have not tested Borg 1.4 personally since Debian ships with 1.2.4.

**Comment by hgy59** [comment]
> I am ready to create documentation should this get merged. Just need to know where is the best place to do so.

package documentation are located in the wiki and linked to https://github.com/SynoCommunity/spksrc/wiki/Package-Documentation-Index (and the side panel).

**Comment by hgy59** [comment]
> @hgy59 no reason. I have not tested Borg 1.4 personally since Debian ships with 1.2.4.

updates done.

**Comment by mreid-tt** [comment]
> updates done.

Changelog could probably do with an update too...

**Comment by tony1661** [comment]
@hgy59 thanks for considering my PR. I've added documentation for [Borgmatic](https://github.com/SynoCommunity/spksrc/wiki/Borgmatic)

**Review by th0ma7 — APPROVED** [review]
LGTM and timely considering other upcoming changes to the python wheel build process.

**Comment by tony1661** [comment]
I noticed that qoriq-6.2.4 is failing. Is that something to be concerned about?

**Comment by mreid-tt** [comment]
Can you review the change log? With the two commits for this PR, the change log may not be accurate.

**Comment by th0ma7** [comment]
> I noticed that qoriq-6.2.4 is failing. Is that something to be concerned about?

This is being tracked through https://github.com/SynoCommunity/spksrc/issues/6297 and fix found and being worked on at https://github.com/SynoCommunity/spksrc/pull/6307.

This may in turn delay for a few days the publishing of this PR once merged so qoriq fix can be reviewed and merged as well.

**Comment by hgy59** [comment]
> Can you review the change log? With the two commits for this PR, the change log may not be accurate.

@mreid-tt do you mean the _commit comments_ with _change log_? 
This is never a problem since it can be edited on "squash and merge" _(and I always edit the commit history to remove obsolete entries and some empty lines)_

File: spk/borgbackup/Makefile [hunk 1]
```diff
@@ -30,7 +30,7 @@ DESCRIPTION = Deduplicating backup program with compression and authenticated en
 
 DISPLAY_NAME = Borg
 STARTABLE = no
-CHANGELOG = "1. Update borg to v1.2.6.<br/>2. Update borgmatic to v1.8.2.<br/>3. Update emborg to v1.37.<br/>4. Migrate to python 3.11."
```
Comment (mreid-tt) [inline spk/borgbackup/Makefile:33]: @hgy59, this is what I meant by the change log

**Review by mreid-tt — APPROVED** [review]

**Comment by mreid-tt** [comment]
I'm not familiar with Borgmatic, so I can't comment on this directly. However, I came across this issue in the Discord channel and asked the user to submit a bug report. If anyone has insights or a way to reproduce the problem, it might be worth investigating further.

> Since the last update of Borg, I can't run borgmatic anymore. I'm on DS214play (i686) which I think is 32 bits. When I try to do, for instance, a borgmatic -c borgmatic.yml list I get a Traceback, which ends up in:
> ```
>   File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/ruamel/yaml/main.py", line 529, in init
>     CParser.init(selfx, stream)
>   File "_ruamel_yaml.pyx", line 302, in ruamel.yaml.clib.ruamelyaml.CParser.__init
> TypeError: a string or stream input is required
> 
> ```
> I tried to uninstall borg and python3, then reinstall, but I have the same error.

**Comment by tony1661** [comment]
@mreid-tt I'd ask the user to verify their config is correct since sometimes borgmatic has changes to syntax that breaks old configs

**Comment by jfangmann** [comment]
> I'm not familiar with Borgmatic, so I can't comment on this directly. However, I came across this issue in the Discord channel and asked the user to submit a bug report. If anyone has insights or a way to reproduce the problem, it might be worth investigating further.
> 
> > Since the last update of Borg, I can't run borgmatic anymore. I'm on DS214play (i686) which I think is 32 bits. When I try to do, for instance, a borgmatic -c borgmatic.yml list I get a Traceback, which ends up in:
> > ```
> >   File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/ruamel/yaml/main.py", line 529, in init
> >     CParser.init(selfx, stream)
> >   File "_ruamel_yaml.pyx", line 302, in ruamel.yaml.clib.ruamelyaml.CParser.__init
> > TypeError: a string or stream input is required
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I tried to uninstall borg and python3, then reinstall, but I have the same error.

I am facing the same problem on my DS214play since the last update. Based on the suggested solution from @tony1661 I tried to create a sample configuration an diff with my existing one. While generating via `borgmatic config generate --destination ./config.yaml` I got this error:

```
Generating a configuration file at: ./config.yaml
Traceback (most recent call last):
  File "/usr/local/bin/borgmatic", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/commands/borgmatic.py", line 940, in main
    list(
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/commands/borgmatic.py", line 726, in collect_highlander_action_summary_logs
    borgmatic.actions.config.generate.run_generate(
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/actions/config/generate.py", line 25, in run_generate
    borgmatic.config.generate.generate_sample_configuration(
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/config/generate.py", line 292, in generate_sample_configuration
    schema = ruamel.yaml.YAML(typ='safe').load(open(schema_filename))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/ruamel/yaml/main.py", line 451, in load
    return constructor.get_single_data()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/ruamel/yaml/constructor.py", line 114, in get_single_data
    node = self.composer.get_single_node()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "_ruamel_yaml.pyx", line 701, in ruamel.yaml.clib._ruamel_yaml.CParser.get_single_node
  File "_ruamel_yaml.pyx", line 902, in ruamel.yaml.clib._ruamel_yaml.CParser._parse_next_event
  File "_ruamel_yaml.pyx", line 919, in ruamel.yaml.clib._ruamel_yaml.input_handler
TypeError: a string value is expected
```

There seems to be a similar bug report from a reddit-user [here](https://www.reddit.com/r/synology/comments/1gtjw4l/borgmatic_stopped_working_after_python_upgrade/).

Thanks for any help in advance! :)

**Comment by tony1661** [comment]
@jfangmann can you provide your borgmatic config?

**Comment by jfangmann** [comment]
Hey @tony1661, thanks for the quick reply! Of course, here it is. I deleted all the comment-lines for better readability. The `patterns_from` statement is considered as  _experimental_ but was working fine before the update.

```
repositories:
    - path: ssh://[...]
      label: borgbase
# Read include/exclude patterns from one or more separate named files,
# one pattern per line. Note that Borg considers this option
# experimental. See the output of "borg help patterns" for more
# details.
patterns_from:
    - /etc/borgmatic/lightroom_patterns
encryption_passphrase: "[...]"
ssh_command: ssh [...]
keep_daily: 7
keep_weekly: 4
keep_monthly: 6
before_backup:
    - echo "Starting a backup."
after_backup:
    - echo "Finished a backup."
```

**Comment by jfangmann** [comment]
I wasn't able to test the config on another system, yet. But I will do so and let you know. But because the config generation fails, I tested this on my DS as well as a RPi running borgmatic via docker:

<details>
<summary>RPi output</summary>

```
jfangmann@rpi3:/opt/borgmatic $ docker exec -it borgmatic /bin/bash
db4fa901eee9:/# borgmatic --version
1.9.3
db4fa901eee9:/# borgmatic config generate --destination ./config.yaml
Generating a configuration file at: ./config.yaml

This includes all available configuration options with example values, the few
required options as indicated. Please edit the file to suit your needs.

If you ever need help: https://torsion.org/borgmatic/#issues

summary:
Generate successful
db4fa901eee9:/#
```
</details>

<details>
<summary>DS output</summary>

```
jfangmann@DiskStation:~$ borgmatic --version
1.9.0
jfangmann@DiskStation:~$ borgmatic config generate --destination ./config.yaml
Generating a configuration file at: ./config.yaml
Traceback (most recent call last):
  File "/usr/local/bin/borgmatic", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/commands/borgmatic.py", line 940, in main
    list(
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/commands/borgmatic.py", line 726, in collect_highlander_action_summary_logs
    borgmatic.actions.config.generate.run_generate(
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/actions/config/generate.py", line 25, in run_generate
    borgmatic.config.generate.generate_sample_configuration(
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/borgmatic/config/generate.py", line 292, in generate_sample_configuration
    schema = ruamel.yaml.YAML(typ='safe').load(open(schema_filename))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/ruamel/yaml/main.py", line 451, in load
    return constructor.get_single_data()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/volume1/@appstore/borgbackup/env/lib/python3.11/site-packages/ruamel/yaml/constructor.py", line 114, in get_single_data
    node = self.composer.get_single_node()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "_ruamel_yaml.pyx", line 701, in ruamel.yaml.clib._ruamel_yaml.CParser.get_single_node
  File "_ruamel_yaml.pyx", line 902, in ruamel.yaml.clib._ruamel_yaml.CParser._parse_next_event
  File "_ruamel_yaml.pyx", line 919, in ruamel.yaml.clib._ruamel_yaml.input_handler
TypeError: a string value is expected
jfangmann@DiskStation:~$
```
</details>

So, the same error (or at least a similar one) seems to occur, even if my config isn't involved. Hope this helps and I will try to test the config soon.

**Comment by jfangmann** [comment]
After reviewing the conversation in reddit, I referred to, I noticed the following regarding the python-version on the DS:

```
jfangmann@DiskStation:~$ python --version
Python 3.8.12
jfangmann@DiskStation:~$ python3.11 --version
Python 3.11.10
```

Since I'm not familiar with python or what version is required, I have no idea if this is how it meant to be. @tony1661 Might this be a problem, or is it identical on your systems?

**Comment by hgy59** [comment]
I have executed the command `borgmatic config generate --destination ./config.yaml` on the following systems without any issue:

- virutalDSM (x64) DSM 7.2.2-72806 Update 2
- DS218 (aarch64) DSM 6.2.4-25556 Update 7
- DS115j (armada370) DSM 7.1.1-42962 Update 7

Unfortunately I have no access to a Diskstation with evansport arch (like DS214play), but I tested DS115j with (probably) the same DSM version as the DS214play.
So I guess it must be an issue with the system and not the package.

**Comment by hgy59** [comment]
> After reviewing the conversation in reddit, I referred to, I noticed the following regarding the python-version on the DS:
> 
> ```
> jfangmann@DiskStation:~$ python --version
> Python 3.8.12
> jfangmann@DiskStation:~$ python3.11 --version
> Python 3.11.10
> ```
> 
> Since I'm not familiar with python or what version is required, I have no idea if this is how it meant to be. @tony1661 Might this be a problem, or is it identical on your systems?

@jfangmann the installed package does not use python or python3.11 in the path.
it is installed into a virtual environment and you can inspect the venv configuration with:
```
$ cat /var/packages/borgbackup/target/env/pyvenv.cfg
home = /var/packages/python311/target/bin
include-system-site-packages = true
version = 3.11.10
executable = /volume1/@appstore/python311/bin/python3.11
command = /var/packages/python311/target/bin/python3 -m venv --system-site-packages /volume1/@appstore/borgbackup/env
```

**Comment by hgy59** [comment]
To validate the ruamel.yaml package, I run (successfully) the very basic ruamel example (found at https://yaml.readthedocs.io/en/latest/example/)

```python
$ cat ruyaml_test.py

import sys
from ruamel.yaml import YAML

inp = """\
# example
name:
  # details
  family: Smith   # very common
  given: Alice    # one of the siblings
"""

yaml = YAML()
code = yaml.load(inp)
code['name']['given'] = 'Bob'

yaml.dump(code, sys.stdout)
```

running this example should provide the expected output:
```
$ /var/packages/borgbackup/target/env/bin/python ./ruyaml_test.py
# example
name:
  # details
  family: Smith   # very common
  given: Bob      # one of the siblings
```

If this doesn't work on DS412play either, the ruamel.yaml wheel in the package is corrupted.

**Pull request merged**